cache/rediscache/redicache_test.go

```diff
@@ -0,0 +1,205 @@
+package rediscache_test
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/terranodo/tegola/cache"
+	"github.com/terranodo/tegola/cache/rediscache"
+)
+
+// TestNew will run tests against a local redis instance
+// on 127.0.0.1:6379
+func TestNew(t *testing.T) {
+	if os.Getenv("RUN_REDIS_TESTS") != "yes" {
+		fmt.Println("RUN_REDIS_TESTS not set to 'yes' skipping tests")
+		return
+	}
+
+	testcases := []struct {
+		config map[string]interface{}
+		err    error
+	}{
+		{
+			config: map[string]interface{}{
+				"network":  "tcp",
+				"address":  "127.0.0.1:6379",
+				"password": "",
+				"db":       0,
+				"max_zoom": 0,
+			},
+			err: nil,
+		},
+		{
+			config: map[string]interface{}{},
+			err:    nil,
+		},
+		{
+			config: map[string]interface{}{
+				"address": "127.0.0.1:6000",
+			},
+			err: fmt.Errorf("dial tcp 127.0.0.1:6000: getsockopt: connection refused"),
+		},
+	}
+
+	for i, tc := range testcases {
+		_, err := rediscache.New(tc.config)
+		if err != nil {
+			if tc.err != nil && err.Error() == tc.err.Error() {
+				//	correct error returned
+				continue
+			}
+			t.Errorf("testcase (%v) failed. err: %v", i, err)
```

Comment:
We have been moving our test error reporting to a standardized format. 

The format takes the form of:

` '[' $testcase_number '] ' $test_subject ', expected ' $expected_value ' got ' $gotten_value` 

So, for the above it would look like:
```
t.Errorf("[%v] unexpected err, expected %v got %v", i, tc.err, err)
```

However, moving to the subtest approach is preferred. It's a bit more of a change, you can find an example here: https://github.com/terranodo/tegola/blob/v0.6.0/geom/cmp/cmp_test.go